boards/components/src/temperature_rp2040.rs

```diff
@@ -0,0 +1,59 @@
+use capsules::temperature_rp2040::TemperatureRp4020;
```

Comment:
> Not necessarily opposed to this, but given components are made to foster code reuse across board definitions, is this component required? Are there going to be a multitude of boards using the RP2040?
> 

Several boards use RP2040 like the Sparkfun RP2040 family, Feather RP2040 family and the Ardunio RP2040 board.

> Seeing that this is very similar compared to the `temperature_stm` component, maybe we can instead try to combine these two into a `temperature_adc` component? I realize this might be difficult with how the types work currently.

I agree with this, I suggest we leave this for a future PR as it requires a bit more work.